navigator_engine/api.py

```diff
@@ -19,54 +23,31 @@ def decide():
         }
     ```
     """
+    input_data = json.loads(request.data)
+    graph = load_graph(choose_graph(input_data['data']['url']))
+    data_loader = choose_data_loader(input_data['data']['url'])
+    data = data_loader('url', 'authorization_header', input_data['data'])
+    skip_actions = input_data.get('skipActions', [])
+
+    engine = DecisionEngine(graph, data, skip=skip_actions)
+    engine.decide()
+    del engine.decision['node']
+
     return jsonify({
-        "actions": ["<action_id>", "<action_id>", "<action_id>", "<action_id>"],
-        "skippedActions": ["<action_id>", "<action_id>"],
-        "progress": {
-            "overallProgress": 35,
-            "milestoneListFullyResolved": False,
-            "milestones": [
-                {
-                    "id": "xxx",
-                    "title": "Naomi Data Prep",
-                    "completed": True,
-                    "progress": 100
-                }, {
-                    "id": "yyy",
-                    "title": "Shiny 90 Data Prep",
-                    "completed": False,
-                    "progress": 50
-                }, {
-                    "id": "zzz",
-                    "title": "Update Spectrum",
-                    "completed": False,
-                    "progress": 0
-                }
-            ]
-        },
-        "decision": {
-            "id": "xxx",
-            "content": {
-                "title": "Node Title",
-                "displayHTML": "<p>Lorem Ipsum</p>",
-                "skippable": True,
-                "actionURL": "http://fjelltopp.org"
-            }
-        }
+        "decision": engine.decision,
+        "actions": engine.progress.action_breadcrumbs,
+        "skippedActions": engine.progress.skipped,
+        "progress": engine.progress.report,
     })
 
 
-@api_blueprint.route('/action/<action_id>')
-def action(action_id):
+@api_blueprint.route('/action/<node_id>')
```

Comment:
if it's an action endpoint it should accept only action_ids. Probably should raise an error if non action node id is passed.

Reply:
The client should not worry about anything other than node ids.  This should be kept as node_id I believe.